File: Python/beginner/1541.py
# https://judge.beecrowd.com/pt/problems/view/1541
# Construindo Casas

# Importing math and Sys
from math import sqrt, floor

# Functon to calculate landSize
def calculateLandSize(houseSizeA, houseSizeB, percentage):
    houseArea = houseSizeA * houseSizeB
    totalLandArea = (houseArea * 100) / percentage
    return floor(sqrt(totalLandArea))

# Main Loop
while True:
    try:
        houseSizeA, houseSizeB, percentage = map(int, input().split())
    except EOFError:
        break
    except ValueError:
        break
    
        
    if houseSizeA == 0:
        break
        
    landSize = calculateLandSize(houseSizeA, houseSizeB, percentage)
    print(int(landSize))
    
# By: Fernando Serra
# https://github.com/fernandosserra    

File: Haskell/beginner/1541.hs
-- https://judge.beecrowd.com/pt/problems/view/1541
-- Construindo Casas

import Text.Printf (printf)
import Data.List (unfoldr) -- To break the entry String

-- Main function to calculate the land side
calculateLandSide :: Int -> Int -> Int -> Double
calculateLandSide houseSideA houseSideB percentage =
    let houseArea = fromIntegral (houseSideA * houseSideB) :: Double
        totalLandArea = (houseArea * 100.0) / fromIntegral percentage
    in fromIntegral (floor (sqrt totalLandArea)) -- Usamos floor para arredondar para baixo e então convertemos para Double, pois printf espera um Double

-- Auxiliar to read the entry
readNumbers :: String -> [Int]
readNumbers = map read . words

-- main function
main :: IO ()
main = do
    contents <- getContents
    mapM_ processLine $ lines contents
  where
    processLine line = do
        let parts = readNumbers line
        case parts of
            [houseSideA, houseSideB, percentage] ->
                if houseSideA == 0 then return () -- Stop Point (0 0 0)
                else printf "%.0f\n" (calculateLandSide houseSideA houseSideB percentage)
            _ -> return () -- to avoid empty lines


-- By: Fernando Serra
-- https://github.com/fernandosserra